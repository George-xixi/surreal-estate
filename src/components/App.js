import React, { useState } from "react";
import "../styles/app.css";
import { Routes, Route } from "react-router-dom";
import Navbar from "./Navbar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

const App = () => {
  const [userID, setUserID] = useState("");

  const handleLogin = (response) => {
    setUserID(response.id);
  };

  const handleLogout = () => {
    window.FB.logout();
    setUserID("");
  };

  return (
    <div className="app">
      <Navbar onLogin={handleLogin} userID={userID} onLogout={handleLogout} />
      <Routes>
        <Route path="/" element={<Properties userID={userID} />} />
        <Route path="/add-property" element={<AddProperty />} />
      </Routes>
    </div>
  );
};

export default App;
