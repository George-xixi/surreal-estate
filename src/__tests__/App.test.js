import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders app correctly", () => {
    const { asFragment } = render(<App />);
    const h1Element = screen.getByText("Surreal Estate");

    expect(asFragment()).toMatchSnapshot();
    expect(h1Element).toBeInTheDocument();
  });
});
