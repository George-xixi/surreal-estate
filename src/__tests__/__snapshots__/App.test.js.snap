// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="navbar"
    >
      <svg
        aria-label="Home icon"
        class="home-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m12.223 11.641-.223.22-.224-.22a2.224 2.224 0 0 0-3.125 0 2.13 2.13 0 0 0 0 3.07L12 18l3.349-3.289a2.13 2.13 0 0 0 0-3.07 2.225 2.225 0 0 0-3.126 0z"
        />
        <path
          d="m21.707 11.293-9-9a.999.999 0 0 0-1.414 0l-9 9A1 1 0 0 0 3 13h1v7c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-7h1a1 1 0 0 0 .707-1.707zM18.001 20H6v-9.585l6-6 6 6V15l.001 5z"
        />
      </svg>
      <h1>
        surreal estate
      </h1>
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <a
            href="/"
          >
            view properties
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            href="/add-property"
          >
            add property
          </a>
        </li>
      </ul>
    </div>
    <div
      class="properties"
    >
      <h2
        class="properties__heading"
      >
        Properties
      </h2>
      <div
        class="sidebar"
      >
        <form
          class="sidebar__form"
        >
          <label
            for="search"
          >
            <h3
              class="sidebar__header"
            >
              Search for a property
            </h3>
            <input
              id="search"
              name="search"
              type="text"
              value=""
            />
          </label>
          <button
            type="submit"
          >
            Search
          </button>
        </form>
        <h3
          class="sidebar__header"
        >
          Filter by :
        </h3>
        <div
          class="sidebar__links"
        >
          <div
            class="sidebar__links__city"
          >
            <h4>
              City
            </h4>
            <a
              class="sidebar__link"
              href="/?query={\\"city\\":\\"Manchester\\"}"
            >
              Manchester
            </a>
            <a
              class="sidebar__link"
              href="/?query={\\"city\\":\\"Liverpool\\"}"
            >
              Liverpool
            </a>
            <a
              class="sidebar__link"
              href="/?query={\\"city\\":\\"Leeds\\"}"
            >
              Leeds
            </a>
            <a
              class="sidebar__link"
              href="/?query={\\"city\\":\\"Sheffield\\"}"
            >
              Sheffield
            </a>
          </div>
          <div
            class="sidebar__links__price"
          >
            <h4>
              Price
            </h4>
            <a
              class="sidebar__link"
              href="/?sort={\\"price\\":1}"
            >
              High to low
            </a>
            <a
              class="sidebar__link"
              href="/?sort={\\"price\\":-1}"
            >
              Low to high
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
